File: butterfly/config/src/main/java/it/unipd/dstack/butterfly/config/exception/ConfigurationUndefinedException.java
/**
 * @project:   Butterfly
 * @author:    DStack Group
 * @module:    config
 * @fileName:  ConfigurationUndefinedException.java
 * @created:   2019-03-07
 *
 * --------------------------------------------------------------------------------------------
 * Copyright (c) 2019 DStack Group.
 * Licensed under the MIT License. See License.txt in the project root for license information.
 * --------------------------------------------------------------------------------------------
 *
 * @description:
 * ConfigurationUndefinedException is the exception thrown by AbstractConfigManager implementations when the
 * user attempts to read an undefined configuration property without providing a default value to return.
 */

package it.unipd.dstack.butterfly.config.exception;

public class ConfigurationUndefinedException extends ConfigurationException {
    public ConfigurationUndefinedException(String property) {
        super(ConfigurationUndefinedException.getMessage(property));
    }

    private static String getMessage(String property) {
        return String.format("The property %s is null, but it should have at least a default value.", property);
    }
}

File: user-manager/slack-account-configurator/src/config/errors/ConfigurationUndefinedException.ts
/**
 * @project:   Butterfly
 * @author:    DStack Group
 * @module:    slack-account-configurator
 * @fileName:  ConfigurationUndefinedException.ts
 * @created:   2019-05-07
 *
 * --------------------------------------------------------------------------------------------
 * Copyright (c) 2019 DStack Group.
 * Licensed under the MIT License. See License.txt in the project root for license information.
 * --------------------------------------------------------------------------------------------
 *
 * @description:
 */

import { ConfigurationException } from './ConfigurationException';

export class ConfigurationUndefinedException extends ConfigurationException {
  constructor(property: string) {
    const message = `The property ${property} is null, but it should have at least a default value.`;
    super(message);
  }
}
